.github/workflows/macos-build.yml

```diff
@@ -0,0 +1,574 @@
+name: Build MacOS
+
+on:
+  workflow_dispatch:
+    inputs:
+      macos-publish-arch-limit:
+        description: 'The allowed arches for macos-publish'
+        type: choice
+        default: all
+        options: ["all", "osx-x64", "osx-arm64", "mas-x64", "mas-arm64"]
```

Comment:
Is this just for debugging or do we have a legit long-term usecase for single arch publishing?

Reply:
I was originally hoping we could use these similar to how we use CircleCI right now, to do certain jobs/steps conditionally based on the arch (see example in the Circle config here: https://github.com/electron/electron/blob/main/.circleci/config/base.yml#L2092). I haven't figured out a clean solution with Actions, though, so we can remove this and re-add something similar if/when it works properly.